### PR TITLE
Added User validation and response.js handling responses

### LINKED ISSUE #5: Create a Single function for Joi Validation and use it in Auth and Blog Controller
A function for Joi validation which will check different inputs and will return the response 

```
const data = {
	email: req.body.email,
	password: req.body.password,
};
const check = {
	email: Joi.string().email().required().trim(),
	password: Joi.string().required().min(8).max(32)
};
const { error } = Joi.validate(data, check);
if (error) {
	res.status(400).send({'error':error.details[0].message});	 
}
```

into something like 

```
const resultfromJoi = checkInput(['email','password'],data);
if(resultfromJoi != true) return resultfromJoi;
```

In case of errors, the resultfromJoi should contain the error message

### PR DESCRIPTION
**Pull request by whyDontI**
Fix #5, #7 and #2
Add Response js for global Error or Success messages.
Add Joi middleware and user.validator for validating User inputs.
Use user.validator in auth.js

### CODE REVIEW

**Comment by Swap76** [comment]
Hey @whyDontI,

Don't create pull requests of multiple issues into one.

The segregation will be appreciated ;-)

File: src/middleware/validators/user.validator.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+class UserValidator {
+
+    async user(req, res, next) {
+        const schema = Joi.object().keys({
+            email: Joi.string().email(),
+            password: Joi.string().required()
+        });
+
+        try {
+            let result = await Joi.validate(req.body, schema);
```
Comment (Swap76) [inline src/middleware/validators/user.validator.js]: Will love to know the use of await with joi.validate function

File: src/util/response.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+const logger = require('../lib/logger/bunyan');
```
Comment (Swap76) [inline src/util/response.js]: Hey @whyDontI,

Will love to know how the logging levels are set in bunyan.

How we can use it while running the code

File: src/util/response.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            status: status,
+            message: message
+        }, 200);
+        res.status(status).json({
+            message: message
+        })
+    };
+};
+
+module.exports = new SendResponse();
```
Comment (Swap76) [inline src/util/response.js]: Hey @whyDontI,

Can you add the info mation about using logging levels and SendResponse in the Contribution.md file that will be great

File: src/middleware/validators/user.validator.js (hunk 1 above)
Comment (whyDontI) [reply src/middleware/validators/user.validator.js]: Oh, I was just following my general coding practice. Thre is no need for Async Await. It is already blocking operation.
I'll remove it shortly.

File: src/util/response.js (hunk 2 above)
Comment (whyDontI) [reply src/util/response.js]: I'll add the logResponse function in next the commit. I kind of took it for granted.
The logging level is 'info' so whatever above 'info' will be logged.

File: src/util/response.js (hunk 3 above)
Comment (whyDontI) [reply src/util/response.js]: Sure

**Pull request merged**